python/ray/tune/trial_runner.py

```diff
@@ -724,6 +544,7 @@ def _process_trial(self, trial):
         """
         try:
             result = self.trial_executor.fetch_result(trial)
+            result.update(trial_id=trial.trial_id)
```

Comment:
this code is getting super complicated :) we should document this (in a separate pr..)